feature-libs/quote/root/guards/quote-cart.service.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { Observable, ReplaySubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QuoteCartService {
+  private quoteId: Observable<string> = new ReplaySubject<string>(1);
+  private quoteCartActive: Observable<boolean> = new ReplaySubject<boolean>(1);
```

Comment:
`minor`: Instead of casting this field each time you want to use `ReplaySubject` methods, I propose reversed approach - keep the `ReplaySubject` type and update the stream each time you need, but when you want some method returns an observable that can't be modified, use `asObservable`:

```ts
private quoteCartActive = new ReplaySubject<boolean>(1);

 [...]

 public setQuoteId(quoteId: string): void {
    this.quoteId.next(quoteId);
  }

  public getQuoteId(): Observable<string> {
    return this.quoteId.asObservable();
  }
```